File: src/csharp/pulse/cdm/system/environment/actions/SEChangeEnvironmentalConditions.cs
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

namespace Pulse.CDM
{
  public class SEChangeEnvironmentalConditions : SEEnvironmentAction
  {
    protected SEEnvironmentalConditions environmental_conditions=null;
    protected string                    environmental_conditions_file="";

    public SEChangeEnvironmentalConditions()
    {
    
    }

    public SEChangeEnvironmentalConditions(SEChangeEnvironmentalConditions other)
    {
      Copy(other);
    }

    public override void Clear()
    {
      base.Clear();
      if(environmental_conditions != null)
        environmental_conditions.Clear();
      environmental_conditions_file = null;

    }

    public void Copy(SEChangeEnvironmentalConditions other)
    {
      if (this == other)
        return;
      base.Copy(other);
      if(other.environmental_conditions != null)
        GetEnvironmentalConditions().Copy(other.environmental_conditions);
      environmental_conditions_file = other.environmental_conditions_file;
    }

    public bool IsActive()
    {
      return IsValid();
    }

    public override bool IsValid()
    {
      return HasEnvironmentalConditions() || HasEnvironmentalConditionsFile();
    }
    
    public bool HasEnvironmentalConditions()
    {
      return environmental_conditions != null;
    }
    public SEEnvironmentalConditions GetEnvironmentalConditions()
    {
      if (environmental_conditions == null)
        environmental_conditions = new SEEnvironmentalConditions();
      return environmental_conditions;
    }
    public void InvalidateEnvironmentalConditions()
    {
      environmental_conditions = null;
    }

    public bool HasEnvironmentalConditionsFile()
    {
      return environmental_conditions_file != null && environmental_conditions_file.Length > 0;
    }
    public string GetEnvironmentalConditionsFile()
    {
      return environmental_conditions_file;
    }
    public void SetEnvironmentalConditionsFile(string s)
    {
      environmental_conditions_file = s;
    }

    public new string ToString()
    {
      if (HasEnvironmentalConditionsFile())
        return "Envrioment Configuration:"
            + "\n\tEnvironmentalConditionsFile: " + environmental_conditions_file;
      else if (environmental_conditions != null)
        return "Environment Configuration : " + environmental_conditions.ToString();
      else
        return "Action not specified properly";
    }

  }
}

File: src/java/pulse/cdm/system/environment/actions/SEChangeEnvironmentalConditions.java
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

package pulse.cdm.system.environment.actions;

import pulse.cdm.bind.EnvironmentActions.ChangeEnvironmentalConditionsData;
import pulse.cdm.substance.SESubstanceManager;
import pulse.cdm.system.environment.SEEnvironmentalConditions;

public class SEChangeEnvironmentalConditions extends SEEnvironmentAction
{

  private static final long serialVersionUID = -998387113042088499L;
  protected SEEnvironmentalConditions environmentalConditions;
  protected String                    environmentalConditionsFile;
  
  public SEChangeEnvironmentalConditions()
  {
    this.environmentalConditions=new SEEnvironmentalConditions();
  }
  
  public SEChangeEnvironmentalConditions(SEChangeEnvironmentalConditions other)
  {
    this();
    copy(other);    
  }
  
  public void copy(SEChangeEnvironmentalConditions other)
  {
    if(this==other)
      return;
    super.copy(other);
    this.environmentalConditions.copy(other.environmentalConditions);
    this.environmentalConditionsFile=other.environmentalConditionsFile;
  }
  
  public void reset()
  {
    super.reset();
    this.environmentalConditions.reset();
    this.environmentalConditionsFile="";
  }
  
  public boolean isValid()
  {
    return hasEnvironmentalConditions() || hasEnvironmentalConditionsFile();
  }
  
  public static void load(ChangeEnvironmentalConditionsData src, SEChangeEnvironmentalConditions dst, SESubstanceManager subMgr)
  {
    SEEnvironmentAction.load(src.getEnvironmentAction(), dst);
    switch(src.getOptionCase())
    {
    case ENVIRONMENTALCONDITIONSFILE:
      dst.environmentalConditionsFile = src.getEnvironmentalConditionsFile();
      break;
    case ENVIRONMENTALCONDITIONS:
      SEEnvironmentalConditions.load(src.getEnvironmentalConditions(),dst.getEnvironmentalConditions(),subMgr);
      break;
    default: // do nothing
    }
  }
  public static ChangeEnvironmentalConditionsData unload(SEChangeEnvironmentalConditions src)
  {
    ChangeEnvironmentalConditionsData.Builder dst = ChangeEnvironmentalConditionsData.newBuilder();
    unload(src,dst);
    return dst.build();
  }
  protected static void unload(SEChangeEnvironmentalConditions src, ChangeEnvironmentalConditionsData.Builder dst)
  {
    SEEnvironmentAction.unload(src, dst.getEnvironmentActionBuilder());
    if(src.hasEnvironmentalConditions())
      dst.setEnvironmentalConditions(SEEnvironmentalConditions.unload(src.environmentalConditions));
    else if(src.hasEnvironmentalConditionsFile())
      dst.setEnvironmentalConditionsFile(src.environmentalConditionsFile);
  }
  
  public boolean hasEnvironmentalConditions()
  {
    return this.environmentalConditions!=null;
  }
  public SEEnvironmentalConditions getEnvironmentalConditions()
  {
    return this.environmentalConditions;
  }
  
  public boolean hasEnvironmentalConditionsFile()
  {
    return this.environmentalConditionsFile!=null&&!this.environmentalConditionsFile.isEmpty();
  }
  public String getEnvironmentalConditionsFile()
  {
    return this.environmentalConditionsFile;
  }
  public void setEnvironmentalConditionsFile(String s)
  {
    this.environmentalConditionsFile = s;
  }
  
  public String toString()
  {
    if (environmentalConditions != null)
      return "Environment Configuration : "+environmentalConditions.toString();
    else if(this.hasEnvironmentalConditionsFile())
      return "Envrioment Configuration:"
          + "\n\tEnvironmentalConditionsFile: "+this.environmentalConditionsFile;
    else
      return "Action not specified properly";
  }
}